### PR TITLE
ref(tsdb): Clean up counter data type key generation for Redis.

### PR DESCRIPTION
**Pull request by tkaemming**
This makes the method signature to `make_counter_key` more similar to the signature of `make_key` for consistency, and changes the return type of `make_counter_key` to return the a tuple of `(hash key, hash field)` for additional clarity about how these are structured internally.

### CODE REVIEW

**Comment by tkaemming** [comment]
This is marginally slower in a few cases (pushes more work into the loop body) but I think the readability win is worth it.

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Review by mitsuhiko — DISMISSED** [review]

File: tests/sentry/tsdb/test_redis.py [hunk 1]
```diff
@@ -35,11 +35,11 @@ def tearDown(self):
...
     def test_make_counter_key(self):
-        result = self.db.make_counter_key(TSDBModel.project, 1368889980, 1)
-        assert result == 'ts:1:1368889980:1'
+        result = self.db.make_counter_key(TSDBModel.project, 1, to_datetime(1368889980), 1)
+        assert result == ('ts:1:1368889980:1', 1)
 
-        result = self.db.make_counter_key(TSDBModel.project, 1368889980, 'foo')
-        assert result == 'ts:1:1368889980:33'
+        result = self.db.make_counter_key(TSDBModel.project, 1, to_datetime(1368889980), 'foo')
+        assert result == ('ts:1:1368889980:46', self.db.get_model_key('foo'))
```
Comment (tkaemming) [inline tests/sentry/tsdb/test_redis.py]: This behavior has changed at this method level but is the same overall result. This used to have [the output of the `get_model_key` provided](https://github.com/getsentry/sentry/blame/c1ce6511ac2de168cb7f60300a2cd5bf3b1723d6/src/sentry/tsdb/redis.py#L219-L224) to generate the hash key, now it's the [*input* to the `get_model_key` function which is called internally](https://github.com/getsentry/sentry/blame/15d8cde319c64d79b4759b81c028da81468e9ba6/src/sentry/tsdb/redis.py#L131).

**Review by tkaemming — COMMENTED** [review]
I didn't realize that `make_counter_key` was tested at this level. 😅

**Review by mitsuhiko — APPROVED** [review]
what can go wrong :)

**Comment by tkaemming** [comment]
🙃

**Pull request merged**